src/pybind/mgr/snap_schedule/module.py

```diff
@@ -92,6 +92,9 @@ def snap_schedule_list(self, path: str,
         except CephfsConnectionException as e:
             return e.to_tuple()
         if not scheds:
+            if format == 'json':
+                output: Dict[str, str] = {}
+                return 0, json.dumps(output), ''
```

Comment:
When there are no schedules this now returns either 0 or an error depending on the `format` option. Is this intentional or should this be consistent?

Reply:
I was a bit hesitant to apply the same to non-json cases but I felt like `ENOENT` made sense. So it's intentional. The JSON output is meant to be consumed by other tools, if the result is empty there is no need for an error IMO, an empty list is suffient.